### ISSUE TITLE
Carthage update pulls in incompatible versions of FirebaseAnalyticsBinary and FirebaseCrashlyticsBinary

### ISSUE THREAD

**Issue by JuliusBahr** [firebase/firebase-ios-sdk#6037]
### [REQUIRED] Step 1: Describe your environment

  * Xcode version: 11.5
  * Firebase SDK version: 6.9
  * Firebase Component: Analytics, Crashlytics
  * Component version: _____
  * Installation method: `Carthage` version 0.35.0

#### Steps to reproduce:
In your Cartfile add:
binary "https://dl.google.com/dl/firebase/ios/carthage/FirebaseAnalyticsBinary.json"
binary "https://dl.google.com/dl/firebase/ios/carthage/FirebaseCrashlyticsBinary.json"

Call carthage update --platform iOS

What happened? How can we make the problem occur?
The downloaded FirebaseAnalyticsBinary and FirebaseCrashlyticsBinary are incompatible with one another and lead to linker errors.

### Workaround
Set an explicit version of the dependency. I used:
binary "https://dl.google.com/dl/firebase/ios/carthage/FirebaseAnalyticsBinary.json" ~> 6.9
binary "https://dl.google.com/dl/firebase/ios/carthage/FirebaseCrashlyticsBinary.json" ~> 6.27

**Comment by ryanwilson** [comment]
Hi @JuliusBahr - sorry for the issues you're seeing. Can you please share what the linker errors are so we can better understand the issue at hand? Thanks!

**Comment by JuliusBahr** [comment]
@ryanwilson I'm sorry I don't have logs anymore. The linker issue was around PromisesObjC.framework. But not the signing issue seen with iOS 13.3.x.
I updated from a pretty old version of Fabric/Crashlytics to the FirebaseCrashlytics.
When I updated FirebaseAnalytics and FirebaseCrashlytics PromisesObjC was *not* updated. That is where I ran into linker issues.
This was only resolved when I explicitly set the versions in Carthage to the latest versions for both FirebaseAnalytics and FirebaseCrashlytics.
Without setting versions I got into a state were FirebaseCrashlytics and FirebaseAnalytics did not create a new version of PromisesObjC.
